app/models/ChannelTest.scala

```diff
@@ -0,0 +1,48 @@
+package models
+
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto.{deriveEnumerationDecoder, deriveEnumerationEncoder}
+
+sealed trait Status
+
+object Status {
+  case object Live extends Status
+  case object Draft extends Status
+  case object Archived extends Status
+
+  implicit val customConfig: Configuration = Configuration.default.withDefaults
+  implicit val statusEncoder = deriveEnumerationEncoder[Status]
+  implicit val statusDecoder = deriveEnumerationDecoder[Status]
+}
+
+sealed trait Channel
+
+object Channel {
+  case object Epic extends Channel
+  case object EpicAMP extends Channel
+  case object EpicAppleNews extends Channel
+  case object EpicLiveblog extends Channel
+  case object EpicHoldback extends Channel
+  case object Banner1 extends Channel
+  case object Banner2 extends Channel
+  case object Header extends Channel
+
+  implicit val customConfig: Configuration = Configuration.default.withDefaults
+  implicit val statusEncoder = deriveEnumerationEncoder[Channel]
+  implicit val statusDecoder = deriveEnumerationDecoder[Channel]
+}
+
+trait ChannelTest[T] {
```

Comment:
The type parameter here is a bit odd. `T` will be the same type, e.g. `case class BannerTest(...) extends ChannelTest[BannerTest]`.
It's necessary for the `withChannel` and `withPriority` methods, which need to specifically return `T` rather than just `ChannelTest`

Reply:
It's a shame we can't share the implementation of the `withChannel` and `withPriority` methods - but I can't think of a way to do that!

Reply:
Yeah - each implementation just calls `.copy(...)`. But `copy` is only available on case classes, and `ChannelTest` has to be a trait